app/src/org/commcare/preferences/HiddenPreferences.java

```diff
@@ -551,4 +552,14 @@ public static void markRawMediaCleanUpComplete() {
         CommCareApplication.instance().getCurrentApp().getAppPreferences()
                 .edit().putBoolean(RAW_MEDIA_CLEANUP_COMPLETE, true).apply();
     }
+
+    public static boolean isPostLoginSyncNeeded() {
+        return CommCareApplication.instance().getCurrentApp().getAppPreferences()
+                .getBoolean(POST_LOGIN_SYNC_NEEDED, false);
+    }
+    public static void setPostLoginSyncNeeded(boolean syncNeeded) {
```

Comment:
@shubham1g5  This is a good point. I wonder if we should include some app info in the process to ensure that we trigger the sync for the right app.

Reply:
Since cases are domain level and can be associated with any of the apps in the domain, I don't think it's possible to direct the sync notification to a specific app unless we build custom rules in HQ messaging engine to associate a particular rule with a specific app. But until we do that we should assume notifications to be non app-specific.